sale_commission_product_criteria/models/commission.py

```diff
@@ -65,7 +65,7 @@ def check_type_change_allowed_moves(self):
 class CommissionItem(models.Model):
     _name = "commission.item"
     _description = "Commission Item"
-    _order = "applied_on, based_on, categ_id desc, id desc"
+    _order = "applied_on, based_on, category_complete_name desc, id desc"
```

Comment:
I don't think you need this change, but change `_order` of the category itself.

Reply:
As I understand it, adding a many2one field in the order evaluates the ID of the related record, but what we need is for it to be ordered by grandchild > child > parent. If we create the child record first, then the parent, and then the grandchild, the order would be incorrect (grandchild > parent > child). By ordering by complete_name (as categories do), we solve the problem. Am I wrong?

Reply:
When adding a many2one, first you get the IDs ordered by their native model order, and then it's applied here, so you don't need to reorder them (this also means 2 queries, but Odoo does it this way for UX convenience).